scripts/UpdateCoreFxCode.ps1

```diff
@@ -0,0 +1,44 @@
+param([string]$CoreFxRepoRoot)
+
+$RepoRoot = Split-Path -Parent $PSScriptRoot
+
+$FilesToCopy = @(
+    "src\System.Net.WebSockets.Client\src\System\Net\WebSockets\ManagedWebSocket.cs",
+    "src\Common\src\System\Net\WebSockets\WebSocketValidate.cs"
+)
+
+if(!$CoreFxRepoRoot) {
+    $CoreFxRepoRoot = "$RepoRoot\..\..\dotnet\corefx"
+}
+
+if(!(Test-Path $CoreFxRepoRoot)) {
+    throw "Could not find CoreFx repo at $CoreFxRepoRoot"
+}
+$CoreFxRepoRoot = Convert-Path $CoreFxRepoRoot
+
+$DestinationRoot = "$RepoRoot\src\Microsoft.AspNetCore.WebSockets.Protocol\ext"
```

Comment:
We're not going to be able to take that until we start mirroring new CoreFX builds on our feeds. So I recommend doing it in a separate change.